pkg/gce-cloud-provider/compute/gce-compute.go

```diff
@@ -36,16 +36,21 @@ type GCECompute interface {
 	GetDiskOrError(ctx context.Context, volumeZone, volumeName string) (*compute.Disk, error)
 	GetAndValidateExistingDisk(ctx context.Context, configuredZone, name, diskType string, reqBytes, limBytes int64) (exists bool, err error)
 	InsertDisk(ctx context.Context, zone string, diskToCreate *compute.Disk) (*compute.Operation, error)
+	CreateSnapshot(ctx context.Context, zone, diskName string, snapshotToCreate *compute.Snapshot) (*compute.Operation, error)
+	GetSnapshotOrError(ctx context.Context, snapshotName string) (*compute.Snapshot, error)
+	DeleteSnapshot(ctx context.Context, name string) (*compute.Operation, error)
 	DeleteDisk(ctx context.Context, zone, name string) (*compute.Operation, error)
 	AttachDisk(ctx context.Context, zone, instanceName string, attachedDisk *compute.AttachedDisk) (*compute.Operation, error)
 	DetachDisk(ctx context.Context, volumeZone, instanceName, volumeName string) (*compute.Operation, error)
 	GetDiskSourceURI(disk *compute.Disk, zone string) string
 	GetDiskTypeURI(zone, diskType string) string
 	WaitForAttach(ctx context.Context, zone, diskName, instanceName string) error
+	WaitAndGetSnapshot(ctx context.Context, snapshotName string) (*compute.Snapshot, error)
```

Comment:
This is a confusing interface to me, what are we waiting for?

Reply:
it is waiting for until snapshot is at least created (snapshot is entering the UPLOADING phase) or it failed to get snapshot. 
After your refactor code is in, this interface is not needed. It would be just a function inside of the cloud provider.

Reply:
ok, are you planning to wait until the cloud provider interface refactor is in?

This PR could potentially go in first, it is up to you